aten/src/ATen/native/cpu/UpSampleKernelAVXAntialias.h

```diff
@@ -315,28 +312,31 @@ void upsample_avx_bilinear_uint8(
             /*align_i32=*/true);
   }
 
-  bool is_rgba = num_channels == 4 && input.is_contiguous(at::MemoryFormat::ChannelsLast);
+  bool needs_unpacking = num_channels == 4 && input.is_contiguous(at::MemoryFormat::ChannelsLast);
 
   at::Tensor buffer_horiz, buffer_vert;
-  if (need_horizontal && !(is_rgba && !need_vertical)) {
+  // Minor optimization: we can avoid allocating an extra buffer if we're performing
+  // horizontal-only or vertical-only interpolation, and if the tensor doesn't
+  // need unpacking
+  if (need_horizontal && !(needs_unpacking && !need_vertical)) {
     buffer_horiz = at::empty({4, yin, xout}, input.options());
   }
-  if (need_vertical && !is_rgba) {
+  if (need_vertical && !needs_unpacking) {
     buffer_vert = at::empty({4, yout, xout}, input.options());
   }
 
-  // TODO: The unpack / pack operations create a copy of the original input and
-  // output tensor. There should be a way to avoid these copies by instead
+  // TODO: The unpack / pack operations create a
+  // copy of the original input and output tensor. There should be a way to avoid these copies by instead
```

Comment:
Was this intentional? Looks worse this way.

Reply:
Will revert that :)